### PR TITLE
Rewrite RaceMenuPage to C++

### PR DESCRIPTION
**Pull request by GnomedDev**

### CODE REVIEW

File: payload/game/ui/page/VsRaceMenuPage.hh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/VsRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/util/Registry.hh [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 #include <Common.hh>
 
+#define ENABLE_MR false
```
Comment (stblr) [inline payload/game/util/Registry.hh:5]: I think this should only control the buttons in the `SingleTopPage` to keep the differences as minimal as possible.

File: payload/game/ui/page/ToRaceMenuPage.hh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "game/util/Registry.hh"
+#if ENABLE_MR
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/ToRaceMenuPage.hh:6]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/ui/page/TaRaceMenuPage.hh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/TaRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/ui/page/ReplayGhostRaceMenuPage.hh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/ReplayGhostRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/ui/page/GhostWatchRaceMenuPage.hh [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/GhostWatchRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/ui/page/BtRaceMenuPage.hh [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/BtRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

File: payload/game/ui/page/AfterTaRaceMenuPage.hh [hunk 8]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "RaceMenuPage.hh"
```
Comment (stblr) [inline payload/game/ui/page/AfterTaRaceMenuPage.hh:3]:
```suggestion
#include "game/ui/page/RaceMenuPage.hh"
```

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/game/util/Registry.hh (hunk 2 above)
Comment (GnomedDev) [reply payload/game/util/Registry.hh:5]: Okay, I'll revert back the guards on other stuff

**Review by stblr — APPROVED** [review]

**Pull request merged**